### PR TITLE
refactor(tooltip): make tooltips use floating-ui package - FE-5498

### PR DESCRIPTION
**Pull request by mkrds**
BREAKING CHANGE: jest versions lower than 27 are no longer supported and might break unit tests

### Proposed behaviour

This PR removes @tippyjs/react package in favor of @floating-ui/react-dom-interactions to handle tooltips

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [x] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [x] UI Tests GitHub check reviewed if required

### Testing instructions

Tooltip regression testing is suggested

### CODE REVIEW

File: src/components/tooltip/tooltip.component.tsx [hunk 1]
```diff
@@ -73,6 +91,127 @@ export const Tooltip = React.forwardRef<HTMLDivElement | null, TooltipProps>(
...
+    flipOverridesRef.current = flipOverrides;
+
+    const arrowReference = useRef<HTMLDivElement | null>(null);
+
+    const calculateOffset = useCallback(
+      (placement) => {
+        let mainAxis = 9;
+
+        if (isPartOfInput) {
+          const inputSizes = {
```
Comment (edleeks87) [inline src/components/tooltip/tooltip.component.tsx:113]: Would it be worth making this a function and moving it out of the scope of the component so it's not instantiated each time a tooltip is rendered? It's very trivial so I'm happy to leave as is if you disagree


```
mainAxis = inputSizes(placement, inputSize);
```

Comment (mkrds) [reply src/components/tooltip/tooltip.component.tsx:113]: I don't think it's worth it to be honest, sounds like a micro optimization

File: src/components/tooltip/tooltip.spec.tsx [hunk 2]
```diff
@@ -26,111 +28,99 @@ function render(props: Partial<TooltipProps> = {}, renderer: any = mount) {
...
-              fontSize: "14px",
-              lineHeight: "1.5rem",
-              fontWeight: "400",
-              backgroundColor: "var(--colorsSemanticNeutral500)",
-            },
-            render().find(StyledTooltipWrapper)
-          );
+          render({ isVisible: true });
+
+          expect(screen.getByRole("tooltip")).toHaveStyle({
```
[the remark this answers is not in the document]
Comment (mkrds) [reply src/components/tooltip/tooltip.spec.tsx:49]: It looks like `toHaveStyleRule` does not work correctly with i.e. `screen.getByRole("tooltip").children[0]` so I'm leaving it the way it is for now

**Review by nineteen88 — APPROVED** [review]

**Review by nineteen88 — DISMISSED** [review]
Been through a couple of times, I'm approving it but not sure I'm best placed to review this one currently. it seems quite a big change from how it originally was, but code seems fine to me.

**Review by edleeks87 — APPROVED** [review]

**Review by nineteen88 — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by nineteen88 — DISMISSED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by nineteen88 — APPROVED** [review]

**Review by DlgSHi — DISMISSED** [review]

**Review by nineteen88 — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**